models/lung_nodule_ct_detection/scripts/evaluator.py

```diff
@@ -117,12 +116,12 @@ def __init__(
         self,
         device: torch.device,
         val_data_loader: Iterable | DataLoader,
-        detector: torch.nn.Module,
+        network: RetinaNetDetector,
         epoch_length: int | None = None,
         non_blocking: bool = False,
         prepare_batch: Callable = detection_prepare_val_batch,
         iteration_update: Callable[[Engine, Any], Any] | None = None,
-        inferer: Inferer | None = None,
+        inferer: RetinaNetInferer | None = None,
```

Comment:
Hi @Can-Zhao ,  thanks for the updates. I'd like to ask a question: with this change, can we get exactly the same inference/validate output?  I suspect the results are different, thus the performance (on val set) be updated in readme too? If the score decreased, may also need to post the comparisons to let users know what are the differences, thus they are able to select which version to use. What do you think?

Reply:
Thank you! Very good point. I run evaluation with the new version, and the result is exactly same with the previous version.

Reply:
Thanks @Can-Zhao . Following up with this question, I also wish to ask about the required GPU memory. Is there any change of it? I'm not familiar with the algorithm, but I guess input the whole 3d image rather than a patch into the inferer may take more GPU memory. Is it still enough to use a 16GB GPU (mentioned in the readme)?
I'm fine with other places, and will approve this PR after resolve this question, thanks!

Reply:
Thank you Yiheng. The GPU memory is the same. This script is just a wrapper. There is no fundamental change here. It did not input the whole 3d image. The GPU still takes a patch

Reply:
I see. Sorry I misunderstood the change. I will leave a comment in the place I feel confused.